pkg/catalogv2/content/content.go

```diff
@@ -179,6 +179,21 @@ func (c *Manager) Icon(namespace, name, chartName, version string) (io.ReadClose
 		return nil, "", err
 	}
 
+	// Check if the system catalog is bundled which means we are at an airgapped environment.
+	// If so, check if the icons are available locally.
+	// If the icons are not available locally, return a nil io.ReadCloser
+	if settings.SystemCatalog.Get() == "bundled" {
+		// If the chart icon is not an HTTP URL and the repository has a commit status,
+		// attempt to get the icon from the git repository.
+		if !isHTTP(chart.Icon) && repo.status.Commit != "" {
+			return git.Icon(namespace, name, repo.status.URL, chart)
+		}
+
+		// If the icon is not available in the git repository, use the fallback icon for airgapped environments.
```

Comment:
Remove this statement

```// If the icon is not available in the git repository, use the fallback icon for airgapped environments.```

Reply:
`		// If the icon is not available in the git repository, use the fallback icon for airgapped environments.
		// the icon will be handled by the UI, as long as we return a nil io.ReadCloser and nil error.`